src/embed/init.ts

```diff
@@ -53,49 +53,62 @@ window['_dlo_rulesFromOpsTeam'] = [
 */
 
 function _dlo_collectRules(): any[] {
-  const results: any[] = [];
-  Object.getOwnPropertyNames(window).forEach((propName) => {
-    if (propName.startsWith('_dlo_rules') === false) return;
-
-    const prop = (window as { [key: string]: any })[propName];
-    if (Array.isArray(prop) === false) {
-      Logger.getInstance().warn(LogMessageType.RuleInvalid,
-        { property: prop, reason: 'Rules list must be an array' });
-      return;
-    }
-
-    prop.forEach((rule: any) => {
-      results.push(rule);
+  try {
+    const startTime = Date.now();
+    const results: any[] = [];
+    Object.getOwnPropertyNames(window).forEach((propName) => {
+      if (propName.startsWith('_dlo_rules') === false) return;
+
+      const prop = (window as { [key: string]: any })[propName];
+      if (Array.isArray(prop) === false) {
+        Logger.getInstance().warn(LogMessageType.RuleInvalid,
+          { property: prop, reason: 'Rules list must be an array' });
+        return;
+      }
+
+      prop.forEach((rule: any) => {
+        results.push(rule);
+      });
     });
-  });
-  return results;
+    Logger.getInstance().info(`DLO rules processing time (ms): ${startTime - Date.now()}`);
```

Comment:
Could we use `debug` versus `info`?  There's some new interest in reducing the amount of ingested events and `debug` would give us a way to keep the feature but not default to the more costly state.

Also, you can pass a `LogContext` as the second arg.  I'm wondering if it makes sense to simply add `timing: number` to `LogContext`.  That way you can make it so we can search greater than or less than for timing entries.